data/keystore/src/main/java/tech/pegasys/artemis/bls/keystore/KeyStorePreConditions.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.bls.keystore;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/** Similar to guava Preconditions but throws our custom exception */
+public class KeyStorePreConditions {
+  public static <T extends @NonNull Object> void checkNotNull(T reference, String errorMessage) {
+    if (reference == null) {
+      throw new KeyStoreValidationException(errorMessage);
+    }
+  }
+
+  public static void checkArgument(boolean expression, String errorMessage) {
+    if (!expression) {
+      throw new KeyStoreValidationException(errorMessage);
+    }
+  }
+}
```

Comment:
This class really shouldn't exist.  A precondition check is there to provide early warning that a developer has stuffed up (ie it should never fail).  If we need to throw a custom exception, then this is really a part of the business logic, not a precondition.  We shouldn't hide that and make it look like a precondition check.

Particularly `checkNotNull` should be a precondition check and just use `Preconditions` from guava - something is very wrong if you're passing null where you shouldn't.  The `checkArgument` cases probably need to be reviewed on a case by case basis depending on whether they're checking user input (which makes it business logic) or fixed input (precondition check).

Reply:
I hear you. Will remove this class and use guava instead.